sympy/holonomic/holonomic.py

```diff
@@ -520,7 +520,7 @@ def _singularics_to_ord(self):
             y0 = []
             for i in range(a):
                 y0.append(S(0))
-            y0 += b
+            y0 += [j * factorial(a + i) for i, j in enumerate(b)]
```

Comment:
I tested it on an example with `a=1` and `len(b)>1`.
```
# a=1 and len(b)=3
In [13]: expr_to_holonomic((x + 1)*exp(-x), x0=-1, lenics=3)
Out[13]: HolonomicFunction((x) + (x + 1)Dx, x), {1: [E, -E, E/2]}

In [14]: expr_to_holonomic((x + 1)*exp(-x), x0=-1, lenics=3)._singularics_to_ord()
Out[14]: HolonomicFunction((x) + (x + 1)Dx, x), f(-1) = 0, f'(-1) = E, f''(-1) = -2*E, f'''(-1) = 3*E

In [15]: f =(x + 1)*exp(-x)

In [16]: for i in range(4):
    print f.subs(x, -1).simplify()
    f = f.diff()
   ....:     
0
E
-2*E
3*E
```
The output seems correct, but may be that's not what you meant to ask?